src/main/java/com/digitaldan/jomnilinkII/Connection.java

```diff
@@ -518,7 +518,7 @@ public Message reqObjectProperties(int objectType, int objectNum, int direction,
 													.build();
 		Message msg = sendAndReceive(reqObjectProperties);
 		if (msg.getMessageType() != Message.MESG_TYPE_OBJ_PROP
-				&& msg.getMessageType() != Message.MESG_TYPE_END_OF_DATA) {
+				|| msg.getMessageType() != Message.MESG_TYPE_END_OF_DATA) {
```

Comment:
Can you explain what the issue is here ?  This code has been around for a very long time and i'm trying to understand what this is fixing, and why it has not shown up before.

Reply:
nm, i see the original code   does make sense at all, i'm not sure why this never bubbled up as an issue before, strange.